### PR TITLE
fix: package publishing an already signed package

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This fixes an issue with attempting to publish an already signed package through the CLI.

## Related Issue

Fixes #N/A

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

**Review by lucasrod16 — CHANGES_REQUESTED** [review]
We need to update the nightly ECR test to require no error when pulling signed Zarf packages without providing a public key since we do not perform signature validation on pull:

https://github.com/defenseunicorns/zarf/blob/09ec9e7f05d75409f8a6bc136d94a112dae098ae/src/test/nightly/ecr_publish_test.go#L68

We should also remove the part where we are pulling the signed package with a provided public key in the ECR test for the same reason that there is no signature validation on pull anymore:

https://github.com/defenseunicorns/zarf/blob/09ec9e7f05d75409f8a6bc136d94a112dae098ae/src/test/nightly/ecr_publish_test.go#L73

File: src/cmd/package.go [hunk 1]
```diff
@@ -410,8 +410,8 @@ func bindRemoveFlags(v *viper.Viper) {
 
 func bindPublishFlags(v *viper.Viper) {
 	publishFlags := packagePublishCmd.Flags()
-	publishFlags.StringVarP(&pkgConfig.PublishOpts.SigningKeyPath, "key", "k", v.GetString(common.VPkgPublishSigningKey), lang.CmdPackagePublishFlagSigningKey)
-	publishFlags.StringVar(&pkgConfig.PublishOpts.SigningKeyPassword, "key-pass", v.GetString(common.VPkgPublishSigningKeyPassword), lang.CmdPackagePublishFlagSigningKeyPassword)
+	publishFlags.StringVar(&pkgConfig.PublishOpts.SigningKeyPath, "signing-key", v.GetString(common.VPkgPublishSigningKey), lang.CmdPackagePublishFlagSigningKey)
+	publishFlags.StringVar(&pkgConfig.PublishOpts.SigningKeyPassword, "signing-key-pass", v.GetString(common.VPkgPublishSigningKeyPassword), lang.CmdPackagePublishFlagSigningKeyPassword)
```
Comment (Noxsios) [inline src/cmd/package.go:414]: note for @Racer159 , will need to call out this breaking change in release notes

File: src/pkg/packager/publish.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ import (
 // Publish publishes the package to a registry
 func (p *Packager) Publish() (err error) {
 	_, isOCISource := p.source.(*sources.OCISource)
-	if isOCISource {
+	if isOCISource && p.cfg.PublishOpts.SigningKeyPath == "" {
```
Comment (Noxsios) [inline src/pkg/packager/publish.go:29]: its a bit more effort, but we _can_ use the filtering behavior of our copy fn to not copy over the existing sig, then pull down the `zarf.yaml`, then re-sign, then push those 2 layers back up, then make a new manifest and push. If we wanted to go for max efficiency that would do the trick, but it does introduce some complexity and needed understanding of the push/pull flow.

**Review by Noxsios — APPROVED** [review]

**Review by lucasrod16 — APPROVED** [review]
lgtm

File: src/pkg/packager/publish.go (hunk 2 above)
Comment (Racer159) [reply src/pkg/packager/publish.go:29]: added a comment to the component layers tech debt - it would be more efficient but wanted to fix things faster

**Pull request merged**